test/common/upstream/health_check_fuzz.cc

```diff
@@ -221,6 +260,215 @@ void TcpHealthCheckFuzz::raiseEvent(const Network::ConnectionEvent& event_type,
+               grpc_respond.grpc_respond_bytes().grpc_respond_unstructured_bytes().data().size());
+           ++i) {
+        std::vector<uint8_t> chunk(
+            grpc_respond.grpc_respond_bytes().grpc_respond_unstructured_bytes().data(i).begin(),
+            grpc_respond.grpc_respond_bytes().grpc_respond_unstructured_bytes().data(i).end());
+        bufferList.push_back(chunk);
+      }
+      ENVOY_LOG_MISC(trace, "Responded with arbitrarily generated bytes which have no structure.");
+
+      // If the interval timer is enabled, that means that the rpc is complete, as decodeData hit a
+      // certain branch that called onRpcComplete(), logically representing a completed rpc call.
+      // Thus, skip the next responses until explicility invoking interval timer as cleanup.
+      for (size_t i = 0; i < bufferList.size() && !test_sessions_[0]->interval_timer_->enabled_;
+           ++i) {
+        bool end_stream_on_data = !has_trailers && i == bufferList.size() - 1;
+        const auto data =
+            std::make_unique<Buffer::OwnedImpl>(bufferList[i].data(), bufferList[i].size());
+        ENVOY_LOG_MISC(trace, "Responded with arbitrary data");
+        test_sessions_[0]->stream_response_callbacks_->decodeData(*data, end_stream_on_data);
+      }
+    }
+    default:
+      break;
+    }
+  }
+
+  // If the interval timer is enabled, that means that the rpc is complete, as decodeData hit a
+  // certain branch that called onRpcComplete(), logically representing a completed rpc call. Thus,
+  // skip responding with trailers until explicility invoking interval timer as cleanup.
+  if (has_trailers && !test_sessions_[0]->interval_timer_->enabled_) {
+    std::unique_ptr<Http::TestResponseTrailerMapImpl> response_trailers =
+        std::make_unique<Http::TestResponseTrailerMapImpl>(
+            Fuzz::fromHeaders<Http::TestResponseTrailerMapImpl>(
+                grpc_respond.grpc_respond_trailers().trailers(), {}, {}));
+
+    ENVOY_LOG_MISC(trace, "Responded trailers {}", *response_trailers.get());
+
+    test_sessions_[0]->stream_response_callbacks_->decodeTrailers(std::move(response_trailers));
+  }
+
+  // This means that the response did not represent a full rpc response.
+  if (!test_sessions_[0]->interval_timer_->enabled_) {
+    return;
+  }
+
+  // Once it gets here the health checker will have called onRpcComplete(), logically representing a
+  // completed rpc call, which blows away client if reuse connection is set to false or the health
+  // checker had a goaway event with no error flag.
+  if (!reuse_connection_ || received_no_error_goaway_) {
+    ENVOY_LOG_MISC(trace, "Creating client and stream after response.");
+    triggerIntervalTimer(true);
+  } else { // Still want to invoke interval timer after every response to not have the fuzzer have
+           // to wait for a triggerIntervalTimer event
+    ENVOY_LOG_MISC(trace, "Creating stream after response.");
+    triggerIntervalTimer(false);
```

Comment:
if you want to minimize code here you can also pass `!reuse_connection_ || receieved_no_error_goaway_` to `triggerIntervalTimer`. 

Moving logs to inside `triggerIntervalTimer` would also clean-up a lot of logs where it's called elsewhere.

Reply:
Moved logs, also switched boolean.